src/unix/async.c

```diff
@@ -83,12 +91,38 @@ int uv_async_send(uv_async_t* handle) {
   expected = 1;
   if (!atomic_compare_exchange_strong(pending, &expected, 2))
     abort();
+#if defined(UV_PREFER_WAIT)
+  if (handle->pending != 1) {
```

Comment:
this introduces a data race

Reply:
I think the general directly seems reasonable, but the `uv_mutex_lock` must occur before the 1->2 transition, since all memory accesses (except the unlock) must occur before that transition

Reply:
Thank you, PR updated